lib/railway_ui_web/views/message_view_helper.ex

```diff
@@ -1,5 +1,16 @@
 defmodule RailwayUiWeb.MessageViewHelper do
-  def decode_message_body(message_body) do
-    inspect(Jason.decode!(message_body))
+  def decode_message_body(%{message_type: message_type, encoded_message: encoded_message}) do
+    %{"encoded_message" => message_to_decode} = Jason.decode!(encoded_message)
+
+    module = message_type
+    |> RailwayIpc.Core.Payload.module_from_type()
+
+    if Code.ensure_compiled?(module) do
+      module
+      |> RailwayIpc.Core.Payload.decode_message(message_to_decode)
+      |> inspect(pretty: true)
+    else
+      "Unable to decode message due to missing message module!"
```

Comment:
I still think that just showing the encoded message is more helpful here. Then some on could look up when it was sent and find the right IPC message to decode and have the encoded message to use but I think this still works!